django/cantusdb_project/main_app/models/chant.py

```diff
@@ -148,3 +149,62 @@ def get_next_folio(folio):
             next_chant = None
 
         return next_chant
+
+    def get_syllabized_text_and_melody(self):
+        """Returns the syllabized text with the melody of a chant. If volpiano is not established return None
+
+        Returns:
+            list: Returns a list of zip_longest objects containing syllabized melody and corresponding
+            syllabized text elements
+        """
+
+        if self.volpiano:
+            syls_melody = syllabize_melody(self.volpiano)
+
+            if self.manuscript_syllabized_full_text:
+                syls_text = syllabize_text(
+                    self.manuscript_syllabized_full_text, pre_syllabized=True
+                )
+            elif self.manuscript_full_text:
+                syls_text = syllabize_text(
+                    self.manuscript_full_text, pre_syllabized=False
+                )
+                syls_text, syls_melody = postprocess(syls_text, syls_melody)
+            elif self.manuscript_full_text_std_spelling:
+                syls_text = syllabize_text(
+                    self.manuscript_full_text_std_spelling, pre_syllabized=False
+                )
+            elif self.incipit:
+                syls_text = syllabize_text(self.incipit, pre_syllabized=False)
+                syls_text, syls_melody = postprocess(syls_text, syls_melody)
+            else:
+                syls_text = [[""]]
+
+            word_zip = align(syls_text, syls_melody)
+            return word_zip
+
+    def get_syllabized_text(self):
```

Comment:
There is a lot of repeat between this function and `get_syllabized_text_and_melody`. Perhaps these can be reworked in some way.

Reply:
Now joined into one function to avoid repetitive code